homeassistant/components/zwave/__init__.py

```diff
@@ -358,92 +486,28 @@ def log_all(signal, value=None):
-            if node.generic not in generic_device_class and \
-               None not in generic_device_class:
-                _LOGGER.debug("node.generic %s not None and in "
-                              "generic_device_class %s",
-                              node.generic, generic_device_class)
-                continue
-            if node.specific not in specific_device_class and \
-               None not in specific_device_class:
-                _LOGGER.debug("node.specific %s is not None and in "
-                              "specific_device_class %s", node.specific,
-                              specific_device_class)
-                continue
-            if value.command_class not in command_class and \
-               None not in command_class:
-                _LOGGER.debug("value.command_class %s is not None "
-                              "and in command_class %s",
-                              value.command_class, command_class)
-                continue
-            if value_type != value.type and value_type is not None:
-                _LOGGER.debug("value.type %s != value_type %s",
-                              value.type, value_type)
+        # Check if this value should be tracked by an existing entity
+        for values in discovered_values:
+            values.check_value(value)
+
+        for schema in DISCOVERY_SCHEMAS:
+            if not check_node_schema(node, schema):
                 continue
-            if value_genre != value.genre and value_genre is not None:
-                _LOGGER.debug("value.genre %s != value_genre %s",
-                              value.genre, value_genre)
+            if not check_value_schema(
+                    value, schema[const.DISC_VALUES][const.DISC_PRIMARY]):
                 continue
 
-            # Configure node
-            _LOGGER.debug("Adding Node_id=%s Generic_command_class=%s, "
-                          "Specific_command_class=%s, "
-                          "Command_class=%s, Value type=%s, "
-                          "Genre=%s as %s", node.node_id,
-                          node.generic, node.specific,
-                          value.command_class, value.type,
-                          value.genre, component)
-            workaround_component = workaround.get_device_component_mapping(
-                value)
-            if workaround_component and workaround_component != component:
-                if workaround_component == workaround.WORKAROUND_IGNORE:
-                    _LOGGER.info("Ignoring device %s due to workaround.",
-                                 "{}.{}".format(component, object_id(value)))
-                    continue
-                _LOGGER.debug("Using %s instead of %s",
-                              workaround_component, component)
-                component = workaround_component
-
-            name = "{}.{}".format(component, object_id(value))
+            name = "{}.{}".format(
```

Comment:
Great catch! I think what will be best is just passing a reference to device_config to ZWaveDeviceEntityValues, then recheck the node config in the equivalent location when creating the new entity.